packages/block/src/helpers.ts

```diff
@@ -75,3 +79,67 @@ export function getDifficulty(headerData: HeaderData): bigint | null {
   }
   return null
 }
+
+/**
+ * Calculates the excess data gas for a post EIP 4844 block given the parent block header.
+ * @param parent header for the parent block
+ * @param newBlobs number of blobs contained in block
+ * @returns the excess data gas for the prospective next block
+ *
+ * Note: This function expects that it is only being called on a valid block as it does not have
+ * access to the "current" block's common instance to verify if 4844 is active or not.
+ */
+export const calcExcessDataGas = (parent: BlockHeader, newBlobs: number) => {
+  if (!parent._common.isActivatedEIP(4844)) {
+    // If 4844 isn't active on header, assume this is the first post-fork block so excess data gas is 0
+    return BigInt(0)
+  }
+  if (parent.excessDataGas === undefined) {
+    // Given 4844 is active on parent block, we expect it to have an excessDataGas field
+    throw new Error('parent header does not contain excessDataGas field')
+  }
+
+  const consumedDataGas = BigInt(newBlobs) * parent._common.param('gasConfig', 'dataGasPerBlob')
+  const targetDataGasPerBlock = parent._common.param('gasConfig', 'targetDataGasPerBlock')
+
+  if (parent.excessDataGas + consumedDataGas < targetDataGasPerBlock) return BigInt(0)
+  else {
+    return parent.excessDataGas + consumedDataGas - targetDataGasPerBlock
+  }
+}
```

Comment:
Not directly suggesting it, but have you considered putting all these three methods `calcExcessDataGas()`, `getDataGasPrice()` and `calcDataFee()` in the Header class itself? Should we consider?

Reply:
I did but given they require exterior inputs (e.g. the parent header), it made more sense to me to keep them as pure utility functions.  I could be persuaded otherwise if you think better to have them attached to the header class.  I don't have a strong commitment on this one.

Reply:
I would have *some* tendency to move over since otherwise chances are higher I guess that they "might get forgotten" since people mostly interact with the main Block class.

And other (class) methods demand the parent header (e.g.) as well? Or is there a structural difference here.

No very strong opinion though.

Reply:
I guess you're right and I missed this.  The `validateGasLimit`, `ethashCanonicalDifficulty`, and other validation methods with `block` do require the parent block/header.  Will bolt these back on to the header class.

Reply:
So I added `calcExcessDataGas` in as a function on the `BlockHeader` class and I think it really doesn't make sense.  We use this and the other data gas helpers in the client and vm blockbuilding classes in a context where we don't have a current header to call like [here in the blockBuilder class](https://github.com/ethereumjs/ethereumjs-monorepo/blob/200fec664235fa6220160f547d49a3f76da826e6/packages/vm/src/buildBlock.ts#L178).

I think I vote to leave them as is.

Reply:
Sure 👍